src/main/scala/com.socrata.geoexport/http/ExportService.scala

```diff
@@ -141,22 +142,43 @@ class ExportService(sodaClient: UnmanagedCuratedServiceClient) extends SimpleRes
       case Some(encoder) =>
         proxy(req, fourByFours) match {
           case Right(responses) =>
-            OK ~> ContentType(encoder.encodedMIME) ~> Stream({out: OutputStream =>
-
-              val streams = responses.map { r => r.inputStream() }
-              Converter.execute(req.resourceScope, streams, encoder, out) match {
-                case Success(s) =>
-                  out.flush()
-                  log.info(s"Finished writing export for ${fxfs}")
-                case Failure(failure) =>
-                  log.error(
-                    "Encountered a fatal error; 200 status was already committed.",
-                    failure
-                  )
-
-              }
-              responses.foreach(_.close())
-            })
+            encoder match {
+              case ShapefileEncoder =>
+                try {
+                  val layerStream: Layers = responses
+                    .map(r => r.inputStream())
+                    .map(is => new SoQLPackIterator(new DataInputStream(is)))
+
+                  val files = ShapefileEncoder.buildFiles(req.resourceScope, layerStream)
+                  OK ~> ContentType(encoder.encodedMIME) ~> Stream({out: OutputStream =>
+                    ShapefileEncoder.streamZip(files, out)
+                    out.flush()
+                    log.info(s"Finished writing export for ${fxfs}")
+                  })
+                } catch {
+                  case e: Exception => {
```

Comment:
did we get something more specific here than `Exception`? I think it threw an IOException or something similar? Let's catch that one rather than the most general one.